router/indexerCallbacks.go

```diff
@@ -15,8 +15,12 @@ import (
 func (r *Router) getStats(request RouterRequest) {
 	var err error
 	stats := new(api.VochainStats)
-	stats.BlockHeight = r.vocapp.Height()
-	stats.BlockTimeStamp = int32(r.vocapp.State.Header(true).Timestamp)
+	height := r.vocapp.Node.BlockStore().Height()
+	stats.BlockHeight = uint32(height)
+	block := r.vocapp.Node.BlockStore().LoadBlockMeta(height)
+	if block != nil {
```

Comment:
Shouldn't we return an error on the request if the block is nil ?

Reply:
Right! I've updated the function to call SendError when block is nil.  I've also added returns after two `r.SendErrors` that were missing them.